File: skyspace666_ui/src/services/PlaneService.js
import http from "../http-common"; 

class PlaneService {
  getAllPlanes(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/plane/planes`, searchDTO);
  }

  get(planeId) {
    return this.getRequest(`/plane/${planeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/plane?field=${matchData}`, null);
  }

  addPlane(data) {
    return http.post("/plane/addPlane", data);
  }

  update(data) {
  	return http.post("/plane/updatePlane", data);
  }
  
  uploadImage(data,planeId) {
  	return http.postForm("/plane/uploadImage/"+planeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PlaneService();

File: src/main/java/com/skyspace666/service/PlaneService.java
package com.skyspace666.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace666.domain.Plane;
import com.skyspace666.dto.PlaneDTO;
import com.skyspace666.dto.PlaneSearchDTO;
import com.skyspace666.dto.PlanePageDTO;
import com.skyspace666.dto.PlaneConvertCriteriaDTO;
import com.skyspace666.service.GenericService;
import com.skyspace666.dto.common.RequestDTO;
import com.skyspace666.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PlaneService extends GenericService<Plane, Integer> {

	List<Plane> findAll();

	ResultDTO addPlane(PlaneDTO planeDTO, RequestDTO requestDTO);

	ResultDTO updatePlane(PlaneDTO planeDTO, RequestDTO requestDTO);

    Page<Plane> getAllPlanes(Pageable pageable);

    Page<Plane> getAllPlanes(Specification<Plane> spec, Pageable pageable);

	ResponseEntity<PlanePageDTO> getPlanes(PlaneSearchDTO planeSearchDTO);
	
	List<PlaneDTO> convertPlanesToPlaneDTOs(List<Plane> planes, PlaneConvertCriteriaDTO convertCriteria);

	PlaneDTO getPlaneDTOById(Integer planeId);







}





